packages/adapters-library/src/adapters/aave-v3/products/rewards/aaveV3RewardsAdapter.ts

```diff
@@ -0,0 +1,400 @@
+import { ethers, getAddress } from 'ethers'
+import { AdaptersController } from '../../../../core/adaptersController'
+import { ZERO_ADDRESS } from '../../../../core/constants/ZERO_ADDRESS'
+import { Chain } from '../../../../core/constants/chains'
+import { CacheToDb } from '../../../../core/decorators/cacheToDb'
+import {
+  MaxMovementLimitExceededError,
+  NotImplementedError,
+} from '../../../../core/errors/errors'
+import { CustomJsonRpcProvider } from '../../../../core/provider/CustomJsonRpcProvider'
+import { logger } from '../../../../core/utils/logger'
+import { Helpers } from '../../../../scripts/helpers'
+import {
+  Erc20ExtendedMetadata,
+  IProtocolAdapter,
+  ProtocolToken,
+} from '../../../../types/IProtocolAdapter'
+import {
+  GetEventsInput,
+  GetPositionsInput,
+  GetTotalValueLockedInput,
+  MovementsByBlock,
+  PositionType,
+  ProtocolAdapterParams,
+  ProtocolDetails,
+  ProtocolPosition,
+  ProtocolTokenTvl,
+  TokenType,
+  Underlying,
+  UnwrapExchangeRate,
+  UnwrapInput,
+} from '../../../../types/adapter'
+import { Erc20Metadata } from '../../../../types/erc20Metadata'
+import { Protocol } from '../../../protocols'
+import {
+  IncentivesContract,
+  IncentivesContract__factory,
+} from '../../contracts'
+import { RewardsClaimedEvent } from '../../contracts/IncentivesContract'
+
+export const AAVE_ICON_URL = 'https://cryptologos.cc/logos/aave-aave-logo.png'
+
+export class AaveV3RewardsAdapter implements IProtocolAdapter {
+  productId = 'rewards'
+  protocolId: Protocol
+  chainId: Chain
+  helpers: Helpers
+
+  private incentivesContract: IncentivesContract
+
+  adapterSettings = {
+    enablePositionDetectionByProtocolTokenTransfer: false,
+    includeInUnwrap: false,
+  }
+
+  private INCENTIVES_CONTRACT_ADDRESSES = {
+    [Chain.Arbitrum]: getAddress('0x929EC64c34a17401F460460D4B9390518E5B473e'),
+    [Chain.Ethereum]: getAddress('0x8164Cc65827dcFe994AB23944CBC90e0aa80bFcb'),
+    [Chain.Fantom]: getAddress('0x929EC64c34a17401F460460D4B9390518E5B473e'),
+    [Chain.Avalanche]: getAddress('0x929EC64c34a17401F460460D4B9390518E5B473e'),
+    [Chain.Bsc]: getAddress('0xC206C2764A9dBF27d599613b8F9A63ACd1160ab4'),
+    [Chain.Base]: getAddress('0xf9cc4F0D883F1a1eb2c253bdb46c254Ca51E1F44'),
+    [Chain.Polygon]: getAddress('0x929EC64c34a17401F460460D4B9390518E5B473e'),
+    [Chain.Optimism]: getAddress('0x929EC64c34a17401F460460D4B9390518E5B473e'),
+    [Chain.Linea]: '', // not deployed on linea
```

Comment:
Should we instead just omit the entry? Using empty string will break on line 87 no? 👇 
```
    this.incentivesContract = IncentivesContract__factory.connect(
      this.INCENTIVES_CONTRACT_ADDRESSES[this.chainId],
      this.provider,
    )
```

Reply:
We only support this adapter on:
- Ethereum
- Polygon
- Avalanche
- Base
- Arbitrum
- Optimism
- Fantom (it's commented, but it's there)

We should remove this line for Linea.